applications/ShapeOptimizationApplication/python_scripts/custom_variable_utilities.py

```diff
@@ -0,0 +1,41 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+def WriteDictionaryDataOnNodalVariable(data, model_part, nodal_variable):
+    for node_id, tmp_gradient in data.items():
+        model_part.Nodes[node_id].SetSolutionStepValue(nodal_variable, 0, tmp_gradient)
+
+# ------------------------------------------------------------------------------
+def ReadNodalVariableToList(model_part, nodal_variable):
+    variable_values_list = []
```

Comment:
Should we preallocate here? Can be done with `[None]*number_of_entries`:
https://stackoverflow.com/questions/537086/reserve-memory-for-list-in-python

Not tested though.

Reply:
I use this [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/EmpireApplication/python_scripts/empire_wrapper.py#L416) for a while already
Never tested the performance benefit but I suspect/hope it to be faster

Reply:
👍